Kitodo/src/main/java/org/goobi/production/export/ExportXmlLog.java

```diff
@@ -105,8 +106,8 @@ public void startExport(Process process, OutputStream os, String xslt) throws IO
             outp.output(doc, os);
             os.close();
 
-        } catch (Exception e) {
-            throw new IOException(e);
+        } catch (IOException | RuntimeException e) {
+            throw new IOException(e.getMessage(), e);
```

Comment:
Is this change here necessary? It is new IOException, not logger.

Reply:
The `Exception` here resolves to `IOException` and `RuntimeException`. It always was like this, it was just hidden, now it is readable.

**Rationale:** Catching `java.lang.Exception` [is a code smell](https://www.lyth.de/bad-smells-beim-exception-handling/#highlighter_300915). It was happening in Java version before 7 (legacy code was developed in Java 6 for years), because you would have to write one catch block for each type of exception to be caugt which makes the code unreadable lengthy. This is why Java 7 introduced [Multi-Catch Exceptions](http://www.oracle.com/technetwork/articles/java/java7exceptions-486908.html). I had some time at the airport so I fixed these mistakes, too.

But you point me on another drawback here which I will fix: There is no need to wrap an `IOException` inside another `IOException`…